### PR TITLE
Tool to measure history store performance

### PR DESCRIPTION
**Pull request by viquezclaudio**
History store:
`cargo run --release --bin nimiq-performance-history-store -- --batches 10 --tpb 700`

Light History store:
`cargo run --release --bin nimiq-performance-history-store -- --batches 100 --tpb 700 --light`

Rounds operation:
For example, 700 batches in total in 10 rounds (70 batches per round)
`cargo run --release --bin nimiq-performance-history-store -- --batches 700 --rounds 10 --tpb 100`

### CODE REVIEW

File: test-utils/src/performance/history-store/main.rs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        println!("Exercising the history store");
+        Box::new(HistoryStore::new(env.clone(), NetworkId::UnitAlbatross))
+            as Box<dyn HistoryInterface + Sync + Send>
+    };
+
+    let rounds = args.rounds.unwrap_or(1);
+
+    history_store_performance(history_store, env, args.tpb, args.batches, rounds, db_file);
+
+    exit(0);
```
Comment (sisou) [inline test-utils/src/performance/history-store/main.rs]: You can delete the temp database file after the benchmark finishes:

```suggestion
    history_store_performance(history_store, env, args.tpb, args.batches, rounds, db_file);
    let _ = fs::remove_dir_all(temp_dir);
    exit(0);
```

**Review by Eligioo — APPROVED** [review]
LGTM, it has some Clippy warnings that should be easy to resolve.

**Review by paberr — COMMENTED** [review]
I have a version of this tool on `pb/history_store_perf`, which will be compatible with the changes in PR #2692 . I suggest merging the history store changes first and I can then cherry pick the necessary commits over to this PR.

**Comment by paberr** [comment]
> I have a version of this tool on `pb/history_store_perf`, which will be compatible with the changes in PR #2692 . I suggest merging the history store changes first and I can then cherry pick the necessary commits over to this PR.

Pushed the cherry-picked commits.

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**